### PR TITLE
Enable Azure RBAC changes

### PR DESCRIPTION
**Pull request by VenkataChalla554**
### Context

https://trello.com/c/3m8PNLG6/937-enable-azure-rbac-deployment-on-all-services

### Changes proposed in this pull request

 Enable Azure RBAC Changes
1 )  Replaced set-arm-environment-variables to set-kubelogin-environment
2 ) kubelogin convert-kubeconfig -l $(if ${GITHUB_ACTIONS},spn,azurecli) Added to get-cluster-credentials
3 ) kubernetes provider updated to get RBAC enabled.
4 ) Chnages done github action to use make get-cluster-credentials

### Checklist

- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -100,7 +102,12 @@ print-infra-secrets: read-tf-config install-fetch-config set-azure-account
 	bin/fetch_config.rb -s azure-key-vault-secret:${key_vault_name}/${key_vault_infra_secret_name} -f yaml
 
 get-cluster-credentials: read-cluster-config set-azure-account ## make <config> get-cluster-credentials [ENVIRONMENT=<clusterX>]
-	az aks get-credentials --overwrite-existing -g ${AZURE_RESOURCE_PREFIX}-tsc-${CLUSTER_SHORT}-rg -n ${AZURE_RESOURCE_PREFIX}-tsc-${CLUSTER}-aks
+ifeq ($(GITHUB_ACTIONS),)
```
Comment (saliceti) [inline Makefile:105]: Can we use the same approach as in the [template](https://github.com/DFE-Digital/teacher-services-cloud/blob/main/templates/new_service/Makefile#L120-L129) for consistency?

File: terraform/aks/providers.tf [hunk 2]
```diff
@@ -13,6 +13,14 @@ provider "kubernetes" {
   client_certificate     = module.cluster_data.kubernetes_client_certificate
   client_key             = module.cluster_data.kubernetes_client_key
   cluster_ca_certificate = module.cluster_data.kubernetes_cluster_ca_certificate
+    dynamic "exec" {
```
Comment (saliceti) [inline terraform/aks/providers.tf:16]: Please check indentation

File: .github/actions/deploy/action.yml [hunk 3]
```diff
@@ -110,7 +110,7 @@ runs:
     - name: K8 setup
       shell: bash
       run: |
-        az aks get-credentials -g ${{ env.cluster_rg }} -n ${{ env.cluster_name }}
+        make ${{ inputs.environment }} get-cluster-credentials  cluster_rg=${{ env.cluster_rg }} cluster_name=${{ env.cluster_name }}
```
Comment (saliceti) [inline .github/actions/deploy/action.yml:113]: cluster_rg and cluster_name should not be there as the values are computed by the make command

**Comment by saliceti** [comment]
Please update the doc with the make command: https://github.com/DFE-Digital/check-childrens-barred-list/blob/d87aa144368be087674b53fda3e86391dee57fb5/docs/aks-cheatsheet.md?plain=1#L65

File: Makefile [hunk 4]
```diff
@@ -16,24 +18,24 @@ install-fetch-config:
 		&& chmod +x bin/fetch_config.rb \
 		|| true
 
-review:
+review:test-cluster
```
Comment (saliceti) [inline Makefile:21]: Please always add a space character after ":"

File: Makefile [hunk 5]
```diff
@@ -2,6 +2,8 @@ TERRAFILE_VERSION=0.8
 ARM_TEMPLATE_TAG=1.1.0
 RG_TAGS={"Product" : "Database of Qualified Teachers"}
 SERVICE_SHORT=ccbl
+cluster_rg=""
```
Comment (saliceti) [inline Makefile:5]: I believe these are not required anymore

File: Makefile (hunk 4 above)
Comment (VenkataChalla554) [reply Makefile]: added space

File: Makefile (hunk 5 above)
Comment (VenkataChalla554) [reply Makefile]: removed unused

File: docs/aks-cheatsheet.md [hunk 6]
```diff
@@ -62,7 +62,7 @@ Get access credentials for a managed Kubernetes cluster (passing the
 resource group and the name):
```
Comment (saliceti) [inline docs/aks-cheatsheet.md:62]: passing the *environment name*

File: docs/aks-cheatsheet.md [hunk 7]
```diff
@@ -62,7 +62,7 @@ Get access credentials for a managed Kubernetes cluster (passing the
...
 
 ```shell
-az aks get-credentials -g s189t01-tsc-ts-rg -n s189t01-tsc-test-aks
+make review get-cluster-credentials
 ```
 
 When you have multiple cluster credentials loaded, you can switch between clusters
```
Comment (saliceti) [inline docs/aks-cheatsheet.md:68]: Please remove the below as people should use make get-cluster-credentials and not kubectl contexts

File: docs/aks-cheatsheet.md (hunk 6 above)
Comment (saliceti) [reply docs/aks-cheatsheet.md:62]: Please remove the "Select account for az:" section as it's taken care when you run get-cluster-credentials

File: docs/aks-cheatsheet.md (hunk 7 above)
Comment (VenkataChalla554) [reply docs/aks-cheatsheet.md]: Fixed document

**Review by saliceti — APPROVED** [review]

**Pull request merged**